### PR TITLE
Refine listeners name

### PR DESCRIPTION
**Pull request by JimMoen**
listener name field use `$type:$proto`
i.e
`mqtt:tcp` `mqtt:ws` etc. no `mqtt:ws:8033` anymore

unified `listen_on` field as `$ip:$port`
i.e.
only listen on a port without ip_addr '8080', `listen_on` field format as `0.0.0.0:8080`
listen on specific ip and port format as `127.0.0.1:1883`

### CODE REVIEW

**Comment by JimMoen** [comment]
see https://github.com/emqx/minirest/pull/98

File: apps/emqx_management/src/emqx_mgmt_cli.erl [hunk 1]
```diff
@@ -503,16 +504,16 @@ listeners(["stop", _Proto, ListenOn]) ->
     end,
     stop_listener(emqx_listeners:find_by_listen_on(ListenOn1), ListenOn1);
 
-listeners(["restart", "http:management"]) ->
+listeners(["restart", "management:http"]) ->
```
Comment (zmstone) [inline apps/emqx_management/src/emqx_mgmt_cli.erl:507]: keep backward compatibility

**Review by zmstone — CHANGES_REQUESTED** [review]
for all API and CLI interfaces, we need to keep backward compatibility

**Comment by zmstone** [comment]
Test ipv6 interface

**Comment by JimMoen** [comment]
listener rename will cause client reconnect.
The impact is too big, close this PR.

**Pull request closed without merging**